README.md

```diff
@@ -30,12 +38,52 @@ services:
 
 ### Environment Variables
-
 The `example_compose.yml` file uses several environment variables. Make sure to define these in a `.env` file in the root directory of your project:
+- **`COMPOSE_PROJECT_NAME`**: Specifies the name of the Docker Compose project, allowing the cron to target jobs in that specific project.
+- **`DEBUG`**: Set to `true` to enable detailed output for debugging purposes.
+- **`CRON_LOG_DIR`**: Defines the directory where cron stores log files for executed jobs, defaulting to `/var/log/cron`.
+
+### Volume Mounts
+- **`/var/run/docker.sock`**: This is used to enable the Docker client inside the container to communicate with the Docker daemon running on the host. Be careful when using this as it provides elevated privileges.
+- **`./logs/cron:/var/log/cron`**: Mount a directory to store the cron logs. Make sure this folder has the appropriate permissions.
+
+## Supercronic Integration
+
+This image uses [Supercronic](https://github.com/aptible/supercronic) to run cron jobs in a more Docker-friendly way. Supercronic provides better logging, fewer overheads, and can be easily integrated with Docker containers, making it a better alternative to the traditional `cron` utility.
+
+## Managing Cron Jobs
+
+The image comes with several scripts to manage cron jobs:
+
+- **`functions.sh`**: Contains helper functions for script operations.
+- **`startup.sh`**: This script runs at container startup to initialize all required settings and start cron jobs.
+- **`update_cron.sh`**: Used for updating cron jobs dynamically without restarting the container.
+
+Make sure these scripts are executable and are properly configured according to your needs.
+
+## Building and Running the Image
+To build the Docker image, use the following command:
+```bash
+docker build -t wikiteq/cron .
+```
+To run the Docker container, use:
+```bash
+docker run -d --name cron wikiteq/cron
+```
+
+Make sure to properly configure environment variables and volume mounts to suit your needs.
+
+## Contributing
+If you want to contribute to this project, please feel free to submit a pull request or open an issue on [GitHub](https://github.com/WikiTeq/docker-cron).
 
-- `COMPOSE_PROJECT_NAME`: if defined, the cron looks for the jobs in this compose project only
-- `DEBUG`: when it is `true`, it outputs more iformation for debugging
-- `CRON_LOG_DIR`: the directory where cron puts the log files of the executed jobs, `/var/log/cron` by default
+## Future Improvements
+- **Non-Root User**: Modify the Dockerfile to run the container as a non-root user to enhance security and reduce potential risks.
+- **Docker Socket Mount**: Consider alternatives to mounting `/var/run/docker.sock` directly, as it exposes the Docker daemon to the container, which could lead to privilege escalation. Using more secure alternatives such as APIs or remote Docker services can help minimize this risk.
```

Comment:
the `docker-cli` sends the API requests to the socket to get events, labels and run commands (cron jobs) inside other docker containers. There is only one alternative - use network instead of the socket file, but it is much less secure.